### ISSUE TITLE
Possibility of empty spaces during rendering

### ISSUE THREAD

**Issue by kiamesdavies** [pdfme/pdfme#633]
### Describe the bug

During the dynamic height calculation, the height of the head is factored in calculating the full height of the table, but during rendering depending on if showhead is false/true, the heading is removed. This causes extra space to be added to the bottom of the table.

### To Reproduce

Use the same template from the playground: https://playground.pdfme.com/form-viewer. However, set the showHeader of the  Items table option to false. Also, reduce the height of the table from 45 to 30 (to accommodate the removed head). Additionally, remove 15 from the “y” position of the “Subtotal” schema.

See attached template here [template.json](https://github.com/user-attachments/files/17788279/template.json)

### Expected behavior

Except the subtotal text to still be close to the table, after hiding the head

### Your Environment

```markdown
- pdfme package(@pdfme/generator or @pdfme/ui):
- pdfme version:
- Operating system:
- Node.js version or Browser name & version:
```

### Your Error Log

```shell
none
```

**Comment by kiamesdavies** [comment]
the fix I had on my local system was to concat [0] to the body rows in getDynamicHeightsForTable when the showHead is false, otherwise just returning body.rows is going to cause duplicate rows during page breaks

**Comment by hand-dot** [comment]
Could you show me the code?

**Comment by kiamesdavies** [comment]
@hand-dot replacing 
https://github.com/pdfme/pdfme/blob/fb13097f9f92b2ac962e0d8730e080244d622146/packages/schemas/src/tables/dynamicTemplate.ts#L20

with something like 
```ts
return schema.showHead ? 
        table.allRows().map((row) => row.height) : 
       [0].concat(table.body.map(row => row.height));
```

This removes the space caused by the header row, and also doesnt cause during page breaks with `schema.__bodyRange`

**Comment by hand-dot** [comment]
handled by https://github.com/pdfme/pdfme/pull/639
This will be included in the next release.

**Comment by hand-dot** [comment]
Released as  5.2.4
Also [playground](https://playground.pdfme.com/) is updated. we can see the fix.

**Comment by hand-dot** [comment]
@kiamesdavies Thank you for reporting this.
